### ISSUE TITLE
Can not compile the code. npm run compile

### ISSUE THREAD

**Issue by workfigure** [ExchangeUnion/xud#474]
I am not be able to compile the xud code. I am using Nodejs version: v8.10.0, npm version: 3.5.2

Error details:
ammanuel@ammanuel-VirtualBox:~/sources/xud$ npm run compile

> xud@1.0.0-prealpha.3 compile /home/ammanuel/sources/xud
> cross-os precompile && tsc && cross-os postcompile

lib/p2p/Pool.ts:501:17 - error TS2459: Type 'string | AddressInfo' has no property 'address' and no string index signature.

501         const { address, port } = this.server!.address();
                    ~~~~~~~


lib/p2p/Pool.ts:501:26 - error TS2459: Type 'string | AddressInfo' has no property 'port' and no string index signature.

501         const { address, port } = this.server!.address();
                             ~~~~

**Comment by offerm** [comment]
Is this directly from Master? have you made any changes?
try also `npm install` before the compile.

I just checked on latest master and I don't have such a problem.

**Comment by sangaman** [comment]
Yes, this suggests to me that the code you're compiling is inconsistent with master. Can you make sure you're on the master branch and try `git pull` then `npm run compile` again? If not, can you paste here the output of `git status`?

**Comment by michael1011** [comment]
I have seen this error before. This happened to me because I installed [`@types/node`](https://www.npmjs.com/package/@types/node) and I solved this by not adding it as a dependency (I don't remember why I wanted to do that and we don't need it anyways).
